whc-shared/src/main/java/com/ibm/whc/deid/shared/pojo/config/masking/ConditionalMaskingProviderConfig.java

```diff
@@ -40,11 +40,19 @@ public void validate() throws InvalidMaskingConfigurationException {
       throw new InvalidMaskingConfigurationException(
           "`maskRuleSet` must be specified with at least one entry");
     }
+    int offset = 0;
     for (ConditionalMaskRuleSet ruleSet : maskRuleSet) {
       if (ruleSet == null) {
-        throw new InvalidMaskingConfigurationException("entry in `maskRuleSet` is null");
+        throw new InvalidMaskingConfigurationException(
+            "entry at offset " + Integer.toString(offset) + " in `maskRuleSet` is null");
       }
-      ruleSet.validate("maskRuleSet");
+      try {
+        ruleSet.validate("maskRuleSet");
+      } catch (InvalidMaskingConfigurationException e) {
+        throw new InvalidMaskingConfigurationException("entry at offset " + Integer.toString(offset)

```

Comment:
We should try to use the WPHnnnnX codes as much as possible in case we ever want to 1) document known errors and solutions (the code is useful as a reference) or 2) translate the errors one day especially when they get returned in the UI.
Seems like the train already left the station and there's error message strings everywhere instead of the logCodes.properties.

Reply:
Discussed with Denis.  Will address in a follow-up change.